### ISSUE TITLE
resource.distribution.present

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-checks#104]
## Description

Check if an entity distribution URL is present and resolvable

## Priority

- FAIR: Required

## Issues
- should every entity distribution URL be checked, or just a representative one be checked (i.e. the first one in the metadata file)

## Procedure
- extract the entity distribution URL (for the first entity only)
- check if the URL is resolvable (HTTP Head request)

**Comment by rsandusky** [comment]
Suggested enhancement: if the first URL does not resolve, and another URL is present, check the next URL to see if it resolves. Repeat until either a URL in the set resolves OR there are no more URLs to be checked.

**Comment by tedhabermann** [comment]
Order is a funny thing in many metadata dialects. Maybe we are talking order of URLs in a particular section or with a particular path... I think link checking is a useful service that repositories can provide (see [DataCite Link Checker](https://support.datacite.org/docs/link-checker)). They include reports on checked links in the DOI metadata. Not sure about NOAA, NASA, or USGS repositories.

I think it would be helpful if presence and working are two different checks...

**Comment by gothub** [comment]
@tedhabermann Separate checks for presence and resolvable makes sense, as the metadata author would get credit for providing the URL, and bonus points (in second check) if it is accessible.

Some would argue that this is too lenient, and that it should be a single check that only passes if the URL is present and resolvable. However, this doesn't allow for a URL that requires a login. The URL could be valid, but doesn't lend itself to easy checking from the quality engine

 For example, the following `linkage` requires a login (i don't have one so am assuming it resolves):

```
        <ns0:MD_DigitalTransferOptions>
          <ns0:onLine>
            <ns0:CI_OnlineResource>
              <ns0:linkage>
                <ns0:URL>https://doi.pangaea.de/10.1594/PANGAEA.889138?format=textfile</ns0:URL>
              </ns0:linkage>
              <ns0:function>
                <ns0:CI_OnLineFunctionCode codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#CI_OnLineFunctionCode" codeListValue="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#CI_OnLineFunctionCode_download">download</ns0:CI_OnLineFunctionCode>
              </ns0:function>
            </ns0:CI_OnlineResource>
          </ns0:onLine>
        </ns0:MD_DigitalTransferOptions>
```

**Comment by gothub** [comment]
@rsandusky thx, checking the URLs in this way sounds like a good, efficient approach - to stop when a resolvable one is found.

**Comment by gothub** [comment]
Note that the resolvability part of this check is currently disabled, until https://redmine.dataone.org/issues/881 is closed and the fix is incorporated into a DataONE CCI release.

**Comment by gothub** [comment]
Resolvable check added in commit dac8345d71434986e75e7d813f2e8f659f002ede
